### PR TITLE
Migration of api from v3 to v5

### PR DESCRIPTION
**Pull request by bharti-dreambits**
Adding test file for retailers api and changing the api from v3 to v4

### CODE REVIEW

File: tests/test_retailer.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+#           "ean": "0000007740407",
+#           "quantity": 1
+#         }
+#       ]
+#     }
+#   ]
+# }
+
+        @urlmatch(path=r'/retailer/orders')
+        def order_by_id_stub(url, request):
```
Comment (karan-dreambits) [inline tests/test_retailer.py:193]: So this is an inner function ?

File: tests/test_retailer.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+            orById = api.orders.get('2K8290LP8')
+            print(orById.orderId)
+
+            assert orById.orderId == "2K8290LP8"
+            assert isinstance(orById.pickUpPoint, bool)
+            assert orById.pickUpPoint == True
+
+            assert orById.orderPlacedDateTime == "2017-02-09T12:39:48+01:00"
+
+            assert orById.shipmentDetails['pickUpPointName'] == "Albert Heijn: UTRECHT"
```
Comment (karan-dreambits) [inline tests/test_retailer.py:213]: Are these tests passing ?

Comment (bharti-dreambits) [reply tests/test_retailer.py:213]: Yes sir

File: tests/test_retailer.py (hunk 1 above)
Comment (bharti-dreambits) [reply tests/test_retailer.py:193]: Yes sir, I was it in test_plaza.py file and I found this way more relatable

File: bol/retailer/api.py [hunk 3]
```diff
@@ -260,9 +274,14 @@ def get(self, page=1):
         return ReturnItems.parse(self.api, resp.text)
 
     def getSingle(self, rmaId):
-        resp = self.request("GET", path=rmaId)
+        resp = self.request("GET", path=str(rmaId))
         return SingleReturnItem.parse(self.api, resp.text)
 
+    def handleReturnItem(self, rmaId, status_reason, qty):
```
Comment (karan-dreambits) [inline bol/retailer/api.py:280]: @bharti-dreambits @devang-dreambits  Is this related to return items fetching ?

File: bol/retailer/models.py [hunk 4]
```diff
@@ -220,15 +262,20 @@ class Meta:
         items_key = "purchasableShippingLabels"
         item_type = Labels
 
-class Fulfilment(Model):
+class Visible(Model):
 
     class Meta:
         pass
 
+class Visibles(ModelList):
```
Comment (karan-dreambits) [inline bol/retailer/models.py:270]: Rename this to VisibleCountriesCodes

File: bol/retailer/models.py [hunk 5]
```diff
@@ -220,15 +262,20 @@ class Meta:
         items_key = "purchasableShippingLabels"
         item_type = Labels
 
-class Fulfilment(Model):
+class Visible(Model):
```
Comment (karan-dreambits) [inline bol/retailer/models.py:265]: Rename this to VisibleCountryCode

File: bol/retailer/models.py [hunk 6]
```diff
@@ -109,31 +109,65 @@ class Meta:
         billingDetails = ModelField(BillingDetails)
 
 
-class Price(Model):
+# class Price(Model):
```
Comment (karan-dreambits) [inline bol/retailer/models.py:112]: Are we going to use this commented model anywhere further ? 
If not then can you remove this ?

**Review by karan-dreambits — CHANGES_REQUESTED** [review]

File: bol/retailer/api.py (hunk 3 above)
Comment (bharti-dreambits) [reply bol/retailer/api.py:280]: Yes sir these changes are related to "returns"

Comment (karan-dreambits) [reply bol/retailer/api.py:280]: how come this are already in this PR before you even started working on it ?

Comment (bharti-dreambits) [reply bol/retailer/api.py:280]: This PR was firstly for v3 to v4. May be because of that we can see the above code in this PR

Comment (karan-dreambits) [reply bol/retailer/api.py:280]: Okay

File: bol/retailer/api.py [hunk 7]
```diff
@@ -127,8 +141,15 @@ def getById(self, process_id):
     def getByIds(self, process_ids):
         if not type(process_ids) is list:
             return {}
-        resp = self.request("POST", params=process_ids)
-        return ProcessStatus.parse(self.api, resp.text)
+        process_id_dict = {}
```
Comment (karan-dreambits) [inline bol/retailer/api.py:144]: The way this is coded looks quite complex and unnecessary extra lines. 

I think we can simplify it to something like this:
```
        process_id_dict = {
               "processStatusQueries" : []
        }
        process_statuses = process_id_dict["processStatusQueries"]
        for process_id in process_ids:
            process_statuses.append({"processStatusId" : process_id})

        resp = self.request("POST", json=process_id_dict)
        return ProcessStatuses.parse(self.api, resp.text)
```

right ?

Comment (bharti-dreambits) [reply bol/retailer/api.py:144]: Yes, you are right

**Pull request merged**